acceptance/suite_test.go

```diff
@@ -0,0 +1,261 @@
+			// Host defines where docker containers run
+			// This is `localhost` for local env, but in CI containers run in a dind service that runs on `docker` host.
+			Host string `default:"localhost"`
+			// Tag defines which version of our apps has to be tested. In local, these are built using `make build-local`,
+			// while in CI those are images pushed to gcr ready to be used in production.
+			Tag string `default:"local"`
+			// Auth is used in CI to be able to pull images from GCR. In local we expect the images to be already built.
+			Auth docker.AuthConfiguration
+		}
+	}
+
+	pool    *dockertest.Pool
+	network *dockertest.Network
+
+	cortexResource      *dockertest.Resource
+	influxProxyResource *dockertest.Resource
+
+	api struct {
+		influx_client influxdb.Client
+		writeAPI      influxdb_api.WriteAPIBlocking
+		promClient    promapi.Client
+		querierClient promv1.API
+	}
+
+	suiteReady time.Time
+}
+
+func (s *Suite) SetupSuite() {
+	t0 := time.Now()
+	s.Require().NoError(envconfig.Process("ACCEPTANCE", &s.cfg))
+	s.Require().NotEmpty(s.cfg.Docker.Tag, "ACCEPTANCE_DOCKER_TAG should not be empty as that would test the latest version. It should be a branch tag or `local` for local testing")
+
+	pool, err := dockertest.NewPool("")
+	s.Require().NoError(err)
+	s.pool = pool
+	s.Require().NoError(s.pool.Retry(s.pool.Client.Ping), "Docker daemon isn't ready")
+
+	s.network = s.createNetwork()
+	s.cortexResource = s.startCortex()
+	s.influxProxyResource = s.startInfluxProxy()
+
+	influx_write_endpoint := fmt.Sprintf("http://%s:%s/", "0.0.0.0", s.influxProxyResource.GetPort("8080/tcp"))
+	influx_client := influxdb.NewClient(influx_write_endpoint, "my-token")
+	write_api := influx_client.WriteAPIBlocking("my-org", "my-bucket")
+	s.api.influx_client = influx_client
+	s.api.writeAPI = write_api
+
+	// Prometheus client and API for verifying that writes occurred as expected
+	s.api.promClient, _ = promapi.NewClient(promapi.Config{
+		Address: fmt.Sprintf("http://%s:%s/api/prom", s.cfg.Docker.Host, s.cortexResource.GetPort("9009/tcp")),
+	})
+	s.api.querierClient = promv1.NewAPI(s.api.promClient)
+
+	s.suiteReady = time.Now()
+	s.T().Logf("Setup complete, took %s", time.Since(t0))
+}
+
+func (s *Suite) TearDownSuite() {
+	if s.T().Failed() {
+		if s.cfg.CI {
+			s.printLogs(s.influxProxyResource)
+		} else {
+			s.T().Logf("Not stopping containers to allow logs inspection")
+			s.T().Logf("Use the following command to stop them:")
+			s.T().Logf(`docker ps --filter "label=acceptance-suite=influx-proxy" -q | xargs docker stop | xargs docker rm`)
+			return
+		}
+	}
+
+	s.NoError(s.cortexResource.Close())
+	s.NoError(s.influxProxyResource.Close())
+	s.NoError(s.network.Close())
+}
+
+func (s *Suite) SetupTest() {
+	s.T().Logf("Test started at %s", time.Now().Format(time.RFC3339Nano))
+}
+
+func (s *Suite) TearDownTest() {
+	s.T().Logf("Test ended at %s", time.Now().Format(time.RFC3339Nano))
+}
+
+func (s *Suite) printLogs(res *dockertest.Resource) {
```

Comment:
eventually (not now), is this something that could be broken out into an acceptance test library?  This seems to have required a lot of code, and it would be nice if we could reduce the amount of boilerplate required for writing these tests.